apps/ejabberd/src/ejabberd_config.erl

```diff
@@ -597,13 +597,22 @@ process_term_for_hosts_and_pools(Term = {Key, _Val}, State) ->
     BKey = atom_to_binary(Key, utf8),
     case get_key_group(BKey, Key) of
         odbc ->
+            ok = check_pools(State#state.odbc_pools),
             lists:foldl(fun(Pool, S) -> process_db_pool_term(Term, Pool, S) end,
                         State, State#state.odbc_pools);
         _ ->
             lists:foldl(fun(Host, S) -> process_host_term(Term, Host, S) end,
                         State, State#state.hosts)
     end.
 
+check_pools([]) ->
+    ?CRITICAL_MSG("Config file invalid: ODBC defined with no pools", []),
+    exit(no_odbc_pools);
+check_pools(Pools) when is_list(Pools) ->
+    ok;
+check_pools(_) ->
+    {error, bad_arg}.
```

Comment:
The convention is to simply use atom `badarg` without underscore. Anyway, this clause is not needed, as this function will fail anyway with a non-list argument and will include the invalid argument value in error description for debugging.